spring-integration-core/src/main/java/org/springframework/integration/IntegrationPatternType.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.2.2
+ */
+public enum IntegrationPatternType {
+
+	message_channel(IntegrationPatternCategory.messaging_channel),
+
+	publish_subscribe_channel(IntegrationPatternCategory.messaging_channel),
+
+	executor_channel(IntegrationPatternCategory.messaging_channel),
+
+	pollable_channel(IntegrationPatternCategory.messaging_channel),
+
+	reactive_channel(IntegrationPatternCategory.messaging_channel),
+
+	null_channel(IntegrationPatternCategory.messaging_channel),
+
+	bridge(IntegrationPatternCategory.messaging_endpoint),
+
+	service_activator(IntegrationPatternCategory.messaging_endpoint),
+
+	outbound_channel_adapter(IntegrationPatternCategory.messaging_endpoint),
+
+	inbound_channel_adapter(IntegrationPatternCategory.messaging_endpoint),
+
+	outbound_gateway(IntegrationPatternCategory.messaging_endpoint),
+
+	inbound_gateway(IntegrationPatternCategory.messaging_endpoint),
+
+	gateway(IntegrationPatternCategory.messaging_endpoint),
+
+	splitter(IntegrationPatternCategory.message_routing),
+
+	transformer(IntegrationPatternCategory.message_transformation),
+
+	header_enricher(IntegrationPatternCategory.message_transformation),
+
+	filter(IntegrationPatternCategory.message_routing),
+
+	content_enricher(IntegrationPatternCategory.message_transformation),
+
+	header_filter(IntegrationPatternCategory.message_transformation),
+
+	claim_check_in(IntegrationPatternCategory.message_transformation),
+
+	claim_check_out(IntegrationPatternCategory.message_transformation),
+
+	aggregator(IntegrationPatternCategory.message_routing),
+
+	resequencer(IntegrationPatternCategory.message_routing),
+
+	barrier(IntegrationPatternCategory.message_routing),
+
+	chain(IntegrationPatternCategory.message_routing),
+
+	scatter_gather(IntegrationPatternCategory.message_routing),
+
+	delayer(IntegrationPatternCategory.message_routing),
+
+	control_bus(IntegrationPatternCategory.system_management),
+
+	router(IntegrationPatternCategory.message_routing),
+
+	recipient_list_router(IntegrationPatternCategory.message_routing);
+
+
+	private final IntegrationPatternCategory patternCategory;
+
+	IntegrationPatternType(IntegrationPatternCategory patternCategory) {
+		this.patternCategory = patternCategory;
+	}
+
+	public IntegrationPatternCategory getPatternCategory() {
+		return this.patternCategory;
+	}
+
+	/**
+	 * The Enterprise Integration Pattern categories.
+	 * Used to indicate which pattern category a target component belongs.
+	 */
+	public enum IntegrationPatternCategory {
+
+		messaging_channel(
+				message_channel,
+				publish_subscribe_channel,
+				executor_channel,
+				pollable_channel,
+				reactive_channel,
+				null_channel),
+
+		messaging_endpoint(
+				service_activator,
+				outbound_channel_adapter,
+				inbound_channel_adapter,
+				outbound_gateway,
+				inbound_gateway,
+				gateway,
+				bridge),
+
+		message_routing(
+				splitter,
+				filter,
+				aggregator,
```

Comment:
ditto

maybe splitter. aggregator, scatter_gather, resequencer in a new category?

Reply:
DITTO